src/main/java/org/sagebionetworks/web/client/widget/table/v2/TotalVisibleResultsWidget.java

```diff
@@ -0,0 +1,91 @@
+package org.sagebionetworks.web.client.widget.table.v2;
+
+import static org.sagebionetworks.repo.model.table.QueryOptions.BUNDLE_MASK_QUERY_COUNT;
+import static org.sagebionetworks.web.client.widget.asynch.AsynchronousProgressWidget.WAIT_MS;
+import static org.sagebionetworks.web.client.widget.table.v2.results.QueryBundleUtils.getDefaultQuery;
+
+import org.sagebionetworks.repo.model.asynch.AsynchronousJobStatus;
+import org.sagebionetworks.repo.model.asynch.AsynchronousResponseBody;
+import org.sagebionetworks.repo.model.table.Dataset;
+import org.sagebionetworks.repo.model.table.QueryBundleRequest;
+import org.sagebionetworks.repo.model.table.QueryResultBundle;
+import org.sagebionetworks.repo.model.table.View;
+import org.sagebionetworks.web.client.PopupUtilsView;
+import org.sagebionetworks.web.client.widget.asynch.AsynchronousJobTracker;
+import org.sagebionetworks.web.client.widget.asynch.UpdatingAsynchProgressHandler;
+import org.sagebionetworks.web.shared.asynch.AsynchType;
+
+import com.google.gwt.user.client.ui.IsWidget;
+import com.google.gwt.user.client.ui.Widget;
+import com.google.inject.Inject;
+
+public class TotalVisibleResultsWidget implements IsWidget {
+
+	private TotalVisibleResultsWidgetView view;
+	private Integer totalNumberOfResults;
+	private Integer totalNumberOfVisibleResults;
+	private PopupUtilsView popupUtils;
+	private AsynchronousJobTracker asyncJobTracker;
+
+	private static final String DATASETS_HELP = "Files may be unavailable because you do not have permission to see them, they have been deleted, or the Dataset has been misconfigured.";
+
+	@Inject
+	public TotalVisibleResultsWidget(TotalVisibleResultsWidgetView view, PopupUtilsView popupUtils, AsynchronousJobTracker asyncJobTracker) {
+		this.view = view;
+		this.popupUtils = popupUtils;
+		this.asyncJobTracker = asyncJobTracker;
+	}
+
+	@Override
+	public Widget asWidget() {
+		return view.asWidget();
+	}
+
+	public void configure(View viewEntity) {
```

Comment:
If this component is reused (reconfigured using different views), you probably want to initialize
setNumberOfHiddenResultsVisible(false) here (since it may show this while the async call is being run to answer this question).